### ISSUE TITLE
mongo: Invalid code created for certain Maps

### ISSUE THREAD

**Issue by scottbessler** [immutables/immutables#209]
Say you have:

```java
public interface IValue
{
    String getRaw();

    Map<String, String> getMeta();

}
```

and
```java
@Immutable
@Mongo.Repository("Tester")
public interface Tester {
	Map<String,? extends List<? extends IValue>> getValues();
}
```

you get the following resultant code that does not compile:
```java
    public Criteria valuesGreaterThan(Map<java.lang.String, ? extends List<? extends IValue>> lower) {
      return valuesIn(Range.greaterThan(lower));
    }

    public Criteria valuesLessThan(Map<java.lang.String, ? extends List<? extends IValue>> upper) {
      return valuesIn(Range.lessThan(upper));
    }

    public Criteria valuesAtMost(Map<java.lang.String, ? extends List<? extends IValue>> upperInclusive) {
      return valuesIn(Range.atMost(upperInclusive));
    }

    public Criteria valuesAtLeast(Map<java.lang.String, ? extends List<? extends IValue>> lowerInclusive) {
      return valuesIn(Range.atLeast(lowerInclusive));
    }
```

**Comment by elucash** [comment]
For some reason it treats `Map` as comparable, obviously a bug, thank you for reporting!

On the other hand it's better to not use wildcards in attribute definitions at all. I don't know if we should make it a warning or an error, but most of the time the _Immutables_ would not give you the code you're expecting.

**Comment by elucash** [comment]
Just released in 2.1.1. Thank you!
